sugondat-chain/node/src/command/export_genesis_metadata.rs

```diff
@@ -0,0 +1,69 @@
+use crate::command::new_partial;
+use sc_client_api::HeaderBackend;
+use sc_service::Configuration;
+use sp_api::{Metadata, ProvideRuntimeApi};
+use sp_core::hexdisplay::HexDisplay;
+use std::fs;
+use std::io::{self, Write};
+use std::path::PathBuf;
+
+/// Export the genesis metadata of the parachain.
+///
+/// This will print the metadata into the specified file or stdout if no file is specified. By
+/// default, the metadata is printed in hex format. Use the `--raw` flag to print the metadata in
+/// binary format.
+///
+/// To use it with subxt, you can use the following commands:
+///
+///     sugondat-node export-genesis-metadata --raw > metadata.bin
+///
+///     subxt codegen --file metadata.bin | rustfmt --edition=2021 --emit=stdout > src/metadata.rs
+#[derive(Debug, clap::Parser)]
+pub struct ExportGenesisMetadataCmd {
+    #[allow(missing_docs)]
+    #[command(flatten)]
+    pub shared_params: sc_cli::SharedParams,
```

Comment:
I don't really like this one, because it makes the CLI output to look like below. Spot the needed commands.

```
Options:
      --chain <CHAIN_SPEC>
          Specify the chain specification.

          It can be one of the predefined ones (dev, local, or staging) or it can be a path to a file with the chainspec
          (such as one exported by the `build-spec` subcommand).

      --dev
          Specify the development chain.

          This flag sets `--chain=dev`, `--force-authoring`, `--rpc-cors=all`, `--alice`, and `--tmp` flags, unless
          explicitly overridden. It also disables local peer discovery (see --no-mdns and --discover-local)

  -d, --base-path <PATH>
          Specify custom base path

  -l, --log <LOG_PATTERN>...
          Sets a custom logging filter (syntax: `<target>=<level>`).

          Log levels (least to most verbose) are `error`, `warn`, `info`, `debug`, and `trace`.

          By default, all targets log `info`. The global log level can be set with `-l<level>`.

          Multiple `<target>=<level>` entries can be specified and separated by a comma.

          *Example*: `--log error,sync=debug,grandpa=warn`. Sets Global log level to `error`, sets `sync` target to debug
          and grandpa target to `warn`.

      --detailed-log-output
          Enable detailed log output.

          Includes displaying the log target, log level and thread name.

          This is automatically enabled when something is logged with any higher level than `info`.

      --disable-log-color
          Disable log color output

      --enable-log-reloading
          Enable feature to dynamically update and reload the log filter.

          Be aware that enabling this feature can lead to a performance decrease up to factor six or more. Depending on
          the global logging level the performance decrease changes.

          The `system_addLogFilter` and `system_resetLogFilter` RPCs will have no effect with this option not being set.

      --tracing-targets <TARGETS>
          Sets a custom profiling filter.

          Syntax is the same as for logging (`--log`).

      --tracing-receiver <RECEIVER>
          Receiver to process tracing messages

          [default: log]

          Possible values:
          - log: Output the tracing records using the log

  -r, --raw
          Write output in binary. Default is to write in hex

  -h, --help
          Print help (see a summary with '-h')

  -V, --version
          Print version
```

I am reluctant to do something about it though.